File: internal/config/paths.go
package config

import (
	"fmt"
	"os"
	"path/filepath"
)

// GetConfigDir returns the path to the TermChess configuration directory.
// It returns ~/.termchess/ or an error if the home directory cannot be determined.
func GetConfigDir() (string, error) {
	homeDir, err := os.UserHomeDir()
	if err != nil {
		return "", fmt.Errorf("failed to get home directory: %w", err)
	}
	return filepath.Join(homeDir, ".termchess"), nil
}

// getConfigFilePath returns the full path to the configuration file.
func getConfigFilePath() (string, error) {
	configDir, err := GetConfigDir()
	if err != nil {
		return "", err
	}
	return filepath.Join(configDir, "config.toml"), nil
}

// SaveGamePath returns the full path to the save game file.
// Exported for testing purposes.
func SaveGamePath() (string, error) {
	configDir, err := GetConfigDir()
	if err != nil {
		return "", err
	}
	return filepath.Join(configDir, "savegame.fen"), nil
}

// GetConfigPath returns the absolute path to the configuration file.
// The config file is stored at ~/.termchess/config.toml
func GetConfigPath() (string, error) {
	homeDir, err := os.UserHomeDir()
	if err != nil {
		return "", err
	}
	return filepath.Join(homeDir, ".termchess", "config.toml"), nil
}

File: rust/crates/config/src/paths.rs
//! Path helpers for the TermChess configuration directory and files.
//!
//! Configuration lives under `~/.termchess/`:
//!   - `config.toml` — TOML settings
//!   - `savegame.fen` — FEN game save

use std::path::PathBuf;

use crate::error::ConfigError;

/// Returns the path to the TermChess configuration directory (`~/.termchess/`).
///
/// Returns an error if the home directory cannot be determined.
pub fn get_config_dir() -> Result<PathBuf, ConfigError> {
    let home = home_dir()?;
    Ok(home.join(".termchess"))
}

/// Returns the full path to the configuration file (`~/.termchess/config.toml`).
pub(crate) fn get_config_file_path() -> Result<PathBuf, ConfigError> {
    Ok(get_config_dir()?.join("config.toml"))
}

/// Returns the full path to the save game file (`~/.termchess/savegame.fen`).
///
/// Exported for testing purposes, mirroring the Go `SaveGamePath`.
pub fn save_game_path() -> Result<PathBuf, ConfigError> {
    Ok(get_config_dir()?.join("savegame.fen"))
}

/// Returns the absolute path to the configuration file (`~/.termchess/config.toml`).
pub fn get_config_path() -> Result<PathBuf, ConfigError> {
    let home = home_dir()?;
    Ok(home.join(".termchess").join("config.toml"))
}

/// Resolves the user's home directory, mirroring Go's `os.UserHomeDir`.
fn home_dir() -> Result<PathBuf, ConfigError> {
    dirs::home_dir().ok_or(ConfigError::HomeDirUnavailable)
}
